src/pytypes/onefuzztypes/job_templates.py

```diff
@@ -0,0 +1,173 @@
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel, Field, root_validator, validator
+
+from .enums import ContainerType, UserFieldOperation, UserFieldType
+from .models import JobConfig, NotificationConfig, TaskConfig, TaskContainers
+from .primitives import File
+from .requests import BaseRequest
+from .responses import BaseResponse
+from .validators import check_template_name
+
+
+class UserFieldLocation(BaseModel):
+    op: UserFieldOperation
+    path: str
+
+
+TemplateUserData = Union[bool, int, str, Dict[str, str], List[str], File]
+
+
+class UserField(BaseModel):
+    name: str
+    type: UserFieldType
+    locations: List[UserFieldLocation]
+    required: bool = Field(default=False)
```

Comment:
Should we validate that `required` is `True` if `default` is `None`?

Reply:
It's reasonable to set the value either way, but if it's not provided to be False.